torchvision/prototype/models/depth/stereo/raft_stereo.py

```diff
@@ -0,0 +1,701 @@
+            )
+
+        _, Cf, Hf, Wf = fmap1.shape
+        coords0 = make_coords_grid(batch_size, Hf, Wf).to(fmap1.device)
+        coords1 = make_coords_grid(batch_size, Hf, Wf).to(fmap1.device)
+
+        depth_predictions = []
+        for _ in range(num_iters):
+            coords1 = coords1.detach()  # Don't backpropagate gradients through this branch, see paper
+            corr_features = self.corr_block(centroids_coords=coords1, corr_pyramid=corr_pyramid)
+
+            depth = coords1 - coords0
+            if self.slow_fast:
+                # Using slow_fast GRU (see paper section 3.4). The lower resolution are processed more often
+                for i in range(1, self.num_level):
+                    # We only processed the smallest i levels
+                    level_processed = [False] * (self.num_level - i) + [True] * i
+                    hidden_states = self.update_block(
+                        hidden_states, contexts, corr_features, depth, level_processed=level_processed
+                    )
+            hidden_states = self.update_block(
+                hidden_states, contexts, corr_features, depth, level_processed=[True] * self.num_level
+            )
+            # Take the largest hidden_state to get the depth
+            hidden_state = hidden_states[0]
+            delta_depth = self.depth_head(hidden_state)
+            # in stereo mode, project depth onto epipolar
+            delta_depth[:, 1] = 0.0
+
+            coords1 = coords1 + delta_depth
+            up_mask = None if self.mask_predictor is None else self.mask_predictor(hidden_state)
+            upsampled_depth = upsample_flow((coords1 - coords0), up_mask=up_mask, factor=self.base_downsampling_ratio)
+            depth_predictions.append(upsampled_depth[:, :1])
+
+        return depth_predictions
+
+
+def _raft_stereo(
+    *,
+    weights: Optional[WeightsEnum] = None,
+    progress: bool = False,
+    shared_encoder_weight: bool = False,
+    # Feature encoder
+    feature_encoder_layers: Tuple[int, int, int, int, int],
+    feature_encoder_strides: Tuple[int, int, int, int],
+    feature_encoder_block: Callable[..., nn.Module],
+    # Context encoder
+    context_encoder_layers: Tuple[int, int, int, int, int],
+    context_encoder_strides: Tuple[int, int, int, int],
+    context_encoder_out_with_blocks: List[bool],
+    context_encoder_block: Callable[..., nn.Module],
+    # Correlation block
+    corr_num_levels: int = 4,
+    corr_radius: int = 4,
+    # Motion encoder
+    motion_encoder_corr_layers: Tuple[int, int],
+    motion_encoder_flow_layers: Tuple[int, int],
+    motion_encoder_out_channels: int,
+    # Update block
+    update_block_hidden_dims: List[int],
+    # Flow Head
+    flow_head_hidden_size: int,
+    # Mask predictor
+    mask_predictor_hidden_size: int,
+    use_mask_predictor: bool,
+    slow_fast: bool,
+    **kwargs,
+):
```

Comment:
I think the only part where we dont really use is `use_mask_predictor`, since on raft stereo they always use one.
There some variable that have same value on `raft_stereo_basic` and `raft_stereo_realtime` , 

i.e.`feature_encoder_layers`, `corr_num_levels`, `corr_radius`, `motion_encoder_corr_layers`, `motion_encoder_flow_layers`, `motion_encoder_out_channels`, `flow_head_hidden_size`, `mask_predictor_hidden_size`
 
but I think they are still worth adding as parameter to `_raft_stereo`

Reply:
OK if there is anything you would like to clip, go ahead. Else mark as resolved.

Reply:
I add default value so users who want to use this function builder directly need not to specify everything.